src/output-model.ts

```diff
@@ -0,0 +1,96 @@
+/// <reference path="./typings/events.d.ts" />
+
+import {IOutput} from './i-output';
+import {IOutputModel} from './i-output-model';
+import {EventEmitter} from 'events';
+
+/**
+ * Provides Jupyter outputs to an output area.
+ *
+ * This class manages the conversion of Jupyter messages to output state, and
+ * handles adding the output state to the output area.
+ */
+export class OutputModel extends EventEmitter implements IOutputModel {
+    private _clear_queued: boolean = false;
+    private _state: IOutput[];
+    
+    /**
+     * Public constructor
+     */
+    public constructor() {
+        super();
+        this._state = [];
+    }
+    
+    /**
+     * State
+     * @return {IOutput[]}
+     */
+    public get state(): IOutput[] {
+        return this._state.slice();
+    }
+    /**
+     * State
+     * @param  {IOutput[]} value new state
+     */
+    public set state(value: IOutput[]) {
+        this.emit('change', value, this._state);
```

Comment:
EventEmitter gets converted from node to browser with browserify quite well. In the semi-far future we can use Object.Observe().